server/base/src/test/java/org/apache/accumulo/server/SetEqualityIteratorTest.java

```diff
@@ -150,8 +154,11 @@ public void testTabletWithOneFile() throws IOException {
     // Asserting the result
     assertEquals(new Key(tabletRow, family), setEqualityIteratorOneFile.getTopKey());
     // The iterator should produce a value that is equal to the expected value on the condition
-    var condition = SetEqualityIterator.createCondition(Collections.singleton(file1),
-        storedTabletFile -> storedTabletFile.getMetadata().getBytes(UTF_8), family);
+    var condition =
+        SetEqualityIterator.createCondition(tmOneFile.getFilesMap().entrySet(), entry -> {
+          return (entry.getKey().getMetadata() + SetEqualityIterator.VALUE_SEPARATOR
+              + entry.getValue().encodeAsString()).getBytes(UTF_8);
+        }, family, true);
```

Comment:
I was curious so I tried messing around with the newly added `concatValue` booleans and all three tests that use `createCondition` pass when  `concatValue` is true OR false. Not sure if that is expected or not. If so, is there a test case that we could add that tests that this field is enforced within `createCondition`?

Reply:
ok, I will have to look at this and add more tests. Thanks!